scripts/aptly/aptly_importer.py

```diff
@@ -0,0 +1,338 @@
+            result = self.run([aptly_type.value, 'search',
+                               '-format={{.Package}}',
+                               name,
+                               '$PackageType (= source)'],
+                              return_all_info=True)
+            if result.returncode == 0:
+                return result.stdout.decode('utf-8').splitlines()
+            # handle no source packages scenario
+            if 'no results' in result.stderr.decode('utf-8'):
+                return []
+            else:
+                self.__error('get_source_packages method', result.stderr.decode('utf-8'))
+
+    def get_snapshots_from_mirror(self, mirror_name):
+        result = []
+        output = check_output('aptly snapshot list', shell=True)
+        for row in output.splitlines():
+            if f"from mirror [{mirror_name}" in row.decode():
+                m = re.findall(r"\[(.*)\]: Snapshot", row.decode())
+                result.append(m[0])
+        return result
+
+    def run(self, cmd=[], fail_on_errors=True, show_errors=True, return_all_info=False):
+        run_cmd = ['aptly']
+        if self.config_file:
+            run_cmd += [f"-config={self.config_file}"]
+        run_cmd += cmd
+        if self.debug:
+            print(f"RUN {' '.join(run_cmd)}")
+        try:
+            r = run(run_cmd, stdout=PIPE, stderr=PIPE)
+        except CalledProcessError as e:
+            if return_all_info:
+                return r
+            return False
+        # if return_all_info is enabled return result object
+        if return_all_info:
+            return r
+
+        if r.returncode == 0:
+            return True
+        else:
+            if show_errors:
+                self.__error(run_cmd, f"{r.stderr.decode('utf-8')}", fail_on_errors)
+            return False
+
+
+class UpdaterConfiguration():
+    def __init__(self, input_file):
+        try:
+            self.config = self.__load_config_file(input_file)
+            self.reprepro2aptly = Reprepro2AptlyFilter()
+
+            self.architectures = self.config['architectures']
+            # source was accepted as a valid architecture to indicate that
+            # source packages need to be download. It is not a valid arch in aptly
+            if 'source' in self.config['architectures']:
+                self.architectures = self.config['architectures'].remove('source')
+            self.architectures = self.config['architectures']
+            self.component = self.config['component']
+            self.filter_formula = self.reprepro2aptly.convert(
+                self.config['filter_formula'])
+            self.method = self.config['method']
+            self.name = self.config['name']
+            self.suites = self.config['suites']
+        except KeyError as e:
+            self.__error(f"{e} key was not found in file {input_file}")
+
+    def __error(self, msg):
+        print(f"Configuration file error: {msg} \n", file=stderr)
+        exit(-1)
```

Comment:
Does it make sense to use a low valued exit code rather than -1/255?

Reply:
Used 1 for generic errors, 2 for config file errors. 7e74b46